### DISCUSSION TITLE
Is there a way to set govanor on performance for gpu and cpu

### DISCUSSION THREAD

**Discussion by WesleyVanNeck** [velvet-os/imagebuilder#248] [General]
Is there a way

**Comment by thenameisluk** [comment]
@WesleyVanNeck 
what do you exactly mean?

**Reply by WesleyVanNeck** [reply]
> @WesleyVanNeck what do you exactly mean?

Well https://ajgupta.github.io/android/2015/01/28/CPU-and-GPU-governors/

**Reply by thenameisluk** [reply]
@WesleyVanNeck 
there is that thing called cpupower shipped with velvet kernel
it might be what you are looking for

not sure, haven't used it

**Reply by WesleyVanNeck** [reply]
> @WesleyVanNeck there is that thing called cpupower shipped with velvet kernel it might be what you are looking for
> 
> not sure, haven't used it

How do I use it

**Reply by thenameisluk** [reply]
@WesleyVanNeck 
this should install app for that
```
sudo apt install cpupower-gui 
```

**Comment by hexdump0815** [comment]
have a look at /sys/devices/system/cpu/cpu0/cpufreq (for each cpu core, but some of them are combined into pools which are also available via /sys/devices/system/cpu/cpufreq )

**Reply by awesomefroggy29** [reply]
what do i change? scaling_max_freq or cpuinfo_max_freq?

**Reply by thenameisluk** [reply]
@awesomefroggy29 
looking at https://github.com/hexdump0815/imagebuilder/blob/main/systems/allwinner_h616/rc-local-additions.txt

it's scaling_max_freq

**Reply by awesomefroggy29** [reply]
not exactly what im looking for, is there a way to overclock past 1703000 hz? im using an oak chromebook

**Reply by thenameisluk** [reply]
@awesomefroggy29 
maybe if you modify device tree source ur kernel uses

should be this file
https://github.com/torvalds/linux/blob/f92f4749861b06fed908d336b4dee1326003291b/arch/arm64/boot/dts/mediatek/mt8173.dtsi#L53
but i really strongly recommend you not to
still it's ur decision and device

**Reply by hexdump0815** [reply]
@awesomefroggy29 - overclocking (if possible at all - some socs block it internally) is not really worth it on those devices ... if something faster is required it might be better to look around for the next better performing system ...

**Reply by awesomefroggy29** [reply]
@thenameisluk so how do i use that file? and just so you know, i am ok with modifying the kernel because i orginally used a usb to boot off of (which worked) and then copied it to an sd when i got one, so i can just reuse that usb and test on the usb, and not my sd.

**Reply by thenameisluk** [reply]
@awesomefroggy29 
well you can just rebuild the kernel
https://github.com/hexdump0815/linux-mainline-mediatek-mt81xx-kernel/blob/master/readme.mt7
with file i showed you modified

then flash it as test so it only boots once (no need for extra medium)
https://github.com/hexdump0815/imagebuilder/blob/main/doc/chromebooks/kernel/installing-a-newer-kernel.md

as hexdump said it really isn't worth it
overclocking won't just make ur device hotter but also unstable (it will be likely to shutdown/freeze randomly)

**Reply by awesomefroggy29** [reply]
i have no idea how to rebuild the kernel with that, so uh... can you help me?
